### PR TITLE
Detect the base commit in extract-patches.sh

### PR DESCRIPTION
**Pull request by dagood**
Small improvement to `extract-patches.sh` to simplify patch workflow:

 1. `git am ../../patches/<repo>/*`
 2. Make changes, fixups, etc.
 3. `../../extract-patches.sh`
 4. Reset inadvertent changes to existing patch files due to client formatting or the new "From" commit hash.

### CODE REVIEW

File: extract-patches.sh [hunk 1]
```diff
@@ -2,6 +2,13 @@
 set -euo pipefail
 IFS=$'\n\t'
 
+getIndexedSubmoduleSha() {
+    (
+        cd "$DIR"
+        git ls-tree HEAD src/cli | awk '{print $3}'
```
Comment (dseefeld) [inline extract-patches.sh]: I don't understand why this is specific to cli?

Comment (dagood) [reply extract-patches.sh]: My bad, it's not. Fixing

File: extract-patches.sh [hunk 2]
```diff
@@ -2,6 +2,13 @@
 set -euo pipefail
 IFS=$'\n\t'
 
+getIndexedSubmoduleSha() {
+    (
+        cd "$DIR"
+        git ls-tree HEAD "src/$REPO_NAME" | awk '{print $3}'
```
Comment (crummel) [inline extract-patches.sh]: I think this will fail for NuGet's nested submodules - `REPO_NAME` will just be `Common`, etc.

Comment (dagood) [reply extract-patches.sh]: Yep, fixed. I don't think we've done nested submodule patches in the build, but no need to prevent that here.

**Comment by dagood** [comment]
@dotnet-bot test OSX10.12 Release
[Assuming just machine or nuget flakiness.](https://ci.dot.net/job/dotnet_source-build/job/release_2.1/job/RHEL7.2_Unshared_Release_prtest/37/)

**Pull request merged**